### PR TITLE
Lykhoyda/migrate workspaces

### PR DESCRIPTION
**Pull request by Lykhoyda**

### CODE REVIEW

File: .github/workflows/lint-squid.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
         key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
 
     - name: install packages
-      run: npm ci
+      run: yarn install --immutable
```
Comment (Tbaut) [inline .github/workflows/lint-squid.yaml:26]: This is for subsquid that only officially supports npm, so we need to keep it that way I'd say.
```suggestion
      run: npm ci
```

File: .github/workflows/lint-squid.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
...
 
     - name: lint
-      run: npm run lint
\ No newline at end of file
+      run: yarn lint
```
Comment (Tbaut) [inline .github/workflows/lint-squid.yaml:29]: same as above
```suggestion
      run: npm run lint
```

File: .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
Comment (Tbaut) [inline .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs:1]: All these files shouldn't be part of the PR. If you pull main, the gitignore should be fine, and you'll need to update the git cache with `git rm -r --cached .`

**Review by Tbaut — COMMENTED** [review]
I'm afraid we're going a dangerous route with yarn for subsquid. Let me check with the team if they are very much against supporting yarn before we move on. I messaged them on their telegram group.

File: .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
Comment (Lykhoyda) [reply .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs:1]: We are using non-zero installs strategy, which works more reliably than zero installs. According to documentation  plugins  folder could be committed to the repo 
https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored

**Review by Tbaut — APPROVED** [review]
Let's do it, thanks for the effort
